src/Lesson5/ShopZD2.java

```diff
@@ -0,0 +1,72 @@
+package Lesson5;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class ShopZD2 implements Shop {
+
+    private Map<Product, Integer> mapProduct = new HashMap<>();
```

Comment:
name it properly, so I will know what is stored in the map:
productToAmount or sth like that